### PR TITLE
Fixed trie sync timeout

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed the double list trie sync implementation timeout if no nodes are received in a specified amount (+ added unit and integration tests)

### CODE REVIEW

File: data/trie/doubleListSync.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type doubleListTrieSyncer struct {
 	handlerID                 string
 	trieSyncStatistics        data.SyncStatisticsHandler
 	lastSyncedTrieNode        time.Time
-	timeoutBetweenCommits     time.Duration
+	timeoutNodeReceived       time.Duration
```
Comment (bogdan-rosianu) [inline data/trie/doubleListSync.go:33]: `receivedNodesTimeout` ? in the entire PR

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: data/trie/doubleListSync.go (hunk 1 above)
Comment (iulianpascalau) [reply data/trie/doubleListSync.go]: done

**Review by bogdan-rosianu — APPROVED** [review]

File: data/trie/doubleListSync_test.go [hunk 2]
```diff
@@ -193,6 +193,23 @@ func TestDoubleListTrieSyncer_StartSyncingCanTimeout(t *testing.T) {
 	require.Equal(t, ErrContextClosing, err)
 }
 
+func TestDoubleListTrieSyncer_StartSyncingTimeoutNoNodesReceived(t *testing.T) {
+	numKeysValues := 10
+	trSource, _ := createInMemoryTrie()
+	addDataToTrie(numKeysValues, trSource)
+	_ = trSource.Commit()
+	roothash, _ := trSource.RootHash()
+	log.Info("source trie", "root hash", roothash)
```
Comment (raduchis) [inline data/trie/doubleListSync_test.go:202]: is the log required?

**Review by raduchis — DISMISSED** [review]

**Review by raduchis — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.
One problem was found but is not critical and will be solved. 
image

**Pull request merged**